pkg/distributor/distributor.go

```diff
@@ -1007,155 +1144,57 @@ func (d *Distributor) push(ctx context.Context, pushReq *push.Request) (*mimirpb
 		return nil, err
 	}
 
+	d.updateReceivedMetrics(req, userID)
```

Comment:
this could be moved below the `return` on `:1150`, but then certain metrics which currently are registered with `0` will not be registered at all anymore which will require me to update the tests, so I didn't do that because I'd prefer to not update the tests at all in this PR to make it easy to see that they all still pass without any changes.